test/suite/test_checkpoint33.py

```diff
@@ -0,0 +1,118 @@
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+from test_cc01 import test_cc_base
+from suite_subprocess import suite_subprocess
+import time
+from wiredtiger import stat
+from wtdataset import SimpleDataSet
+
+# test_checkpoint33.py
+#
+# Test that checkpoint will not skip tables that have available space at the end that can be
+# reclaimed through truncation.
+class test_checkpoint33(test_cc_base, suite_subprocess):
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB,'
+    uri = 'table:test_checkpoint33'
+
+    table_numkv = 10000
+    value_size = 1024
+    value = 'a' * value_size
+
+    def delete(self, timestamp):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            c.set_key(k)
+            self.session.begin_transaction()
+            c.remove()
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(timestamp))
+        c.close()
+
+    def populate(self, timestamp):
+        c = self.session.open_cursor(self.uri, None)
+        for k in range(self.table_numkv):
+            self.session.begin_transaction()
+            c[k] = self.value
+            self.session.commit_transaction("commit_timestamp=" + self.timestamp_str(timestamp))
+        c.close()
+
+    def get_size(self):
+        c = self.session.open_cursor('statistics:' + self.uri, None, None)
+        file_size = c[stat.dsrc.block_size][2]
+        c.close()
+        return file_size
+
+    def evict_all(self):
+        evict_cursor = self.session.open_cursor(self.uri, None, "debug=(release_evict)")
+        self.session.begin_transaction()
+        for k in range(self.table_numkv):
+            evict_cursor.set_key(k)
+            evict_cursor.search()
+            evict_cursor.reset()
+        self.session.rollback_transaction()
+        evict_cursor.close()
+
+    def test_checkpoint33(self):
+        # Pin oldest timestamp 1.
+        self.conn.set_timestamp(f'oldest_timestamp={self.timestamp_str(1)}')
+
+        # Create and populate a table at timestamp 2.
+        self.session.create(self.uri, self.create_params)
+        self.session.checkpoint()
+        self.populate(timestamp=2)
+
+        # Make everything stable at timestamp 3.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(3)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete everything at timestamp 4.
+        self.delete(timestamp=4)
+
+        # Make the deletions stable at timestamp 5.
+        self.conn.set_timestamp(f'stable_timestamp={self.timestamp_str(5)}')
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Evict all pages
+        self.evict_all()
```

Comment:
Can you explain why we are evicting all pages please?

```suggestion
        # Evict all pages.
        self.evict_all()
```

Reply:
Added https://github.com/wiredtiger/wiredtiger/pull/11176/commits/e11564cfcf57b3b12733428dab00adb53dcf6cf4